### PR TITLE
update min macos version to 10.15

### PR DESCRIPTION
**Pull request by shuffle2**
This should be all that's required to make the Updater gracefully tell people on < 10.15 to update.

If we can go higher than 10.15, I'd love to know :)

This will unblock std::filesystem stuff

### CODE REVIEW

**Comment by MayImilae** [comment]
The only change 10.15 does to our device coverage is that we would lose support for pre-trashcan Mac Pros that support Metal. Everything else is unchanged compared to 10.14. So this won't impact our userbase much. If this gives us something useful, then sure.

macOS 11 will take out a lot of devices though, including my 2012 MacBook Pro. That one we should hold off on for as long as we can.

**Comment by pizuz** [comment]
> macOS 11 will take out a lot of devices though, including my 2012 MacBook Pro. That one we should hold off on for as long as we can.

There‘s always OCLP for that, but generally you‘re right. On the other hand, running an unsupported OS isn‘t such a good idea, anyway…

**Comment by OatmealDome** [comment]
Should probably update the README while we're at it. (Wiki + website also need to be updated, but can't do that with a PR.)

10.15 is about the limit of what I'm comfortable updating our minimum to. As far as I'm aware, there isn't anything that we can gain by going higher. Like MayImilae said, hardware support from 10.14 -> 10.15 is nearly identical and older Macs start dropping like flies starting with 11, so we should hold off on that.

**Comment by PEmu2** [comment]
Should the deployment target in BuildMacOSUniversalBinary.py be updated as well?

https://github.com/dolphin-emu/dolphin/blob/master/BuildMacOSUniversalBinary.py#L67

> (Wiki + website also need to be updated, but can't do that with a PR.)

The website FAQ needs to be updated regardless of this PR as it still lists High Sierra as supported. Would be great if these parts of the website were available to be edited on GitHub as it's pretty clear they're not getting much attention these days.

File: Readme.md
Comment (PEmu2) [inline Readme.md:18]: I'd probably still add Catalina here as it's technically the name of the OS.

Comment (JosJuice) [reply Readme.md:18]: I think we should include it. It's not just an internal codename, it's user-facing.

**Review by OatmealDome — APPROVED** [review]
Since there might be a beta releasing around Christmas, perhaps we should merge this after that happens.

**Comment by OatmealDome** [comment]
afaik JMC wants to push another beta to fix a bug in 5.0-17995 (see IRC). Something about the FPS counter colors being wrong with specific settings.

**Comment by PEmu2** [comment]
One thing that might be worth taking into account is that while Catalina runs on most of the same hardware as Mojave there are probably some users who haven't upgraded for the sake of preserving 32-bit app support. I don't have the Dolphin analytics numbers to know if that's actually a practical concern or not. As a rough approximation the Steam hardware and software survey currently has Mojave and earlier as roughly ~7% of the macOS user base.

**Review by OatmealDome — APPROVED** [review]
Talked to JMC, looks like the holiday beta was delayed.

**Comment by team-orangeBlue** [comment]
I am now an angry user.
I hate 10.15. I stick to 10.14.6 on my early 2013 MacBook Pro.
I don't see any real use...

**Comment by OPStellar** [comment]
I am angered by this, too. Not because I don't see the use of 10.15, but because it removes all 32-bit support– drivers which are essential for playing a great deal of Windows games.
What even was the point of this requirement?

**Comment by OatmealDome** [comment]
The primary motivator for us dropping 10.14 support was an effort to modernize and clean up our file handling code by switching over to ``std::filesystem``. Because of how Apple implemented ``std::filesystem`` support, it is not possible to use it on macOS versions prior to 10.15.

When looking at dropping an OS, we look at various other factors as well:

* According to [our opt-in analytics data for the week of 19 May 2022](https://pastebin.com/0AqNQDTK), approximately 1.19% of our userbase was using a Mac running a version of macOS older than 10.15.
* As I mentioned in a previous comment on this PR, Macs that can run 10.14 are able to run 10.15 (though I do understand that 10.14 is being used for 32-bit software support).
* macOS 10.14 is no longer being supported by Apple.

We don't make decisions like this lightly. We do make a best effort to support older macOS versions as long as possible (for example, we still support macOS 10.15), but older OSes will be dropped if they get in the way of progress.

The latest version of Dolphin that supports macOS 10.14 is [5.0-18184](https://dolphin-emu.org/download/dev/master/5.0-18184/), if you wish to use that. Of course, it won't have all the latest features and improvements made since then.

**Pull request merged**
